src/server/pps/server/api_server.go

```diff
@@ -582,49 +582,29 @@ func (a *apiServer) getDatum(ctx context.Context, repo string, commit *pfs.Commi
 	}
 
 	// Check if skipped
-	path := fmt.Sprintf("/%v", datumID)
-	newFile := &pfs.File{
+	stateFile := &pfs.File{
 		Commit: commit,
-		Path:   path,
-	}
-	commitInfo, err := pfsClient.InspectCommit(
-		ctx,
-		&pfs.InspectCommitRequest{
-			Commit: &pfs.Commit{
-				Repo: commit.Repo,
-				ID:   commit.ID,
-			},
-		},
-	)
-	if err != nil {
-		return nil, err
-	}
-	oldFile := &pfs.File{
-		Commit: commitInfo.ParentCommit,
-		Path:   path,
+		Path:   fmt.Sprintf("/%v/skipped", datumID),
 	}
-	resp, err := pfsClient.DiffFile(ctx, &pfs.DiffFileRequest{newFile, oldFile, true})
-	if err != nil {
-		return nil, err
-	}
-	// Datum wasn't added in this commit, so it was skipped
-	if len(resp.NewFiles) == 0 {
+	_, err = pfsClient.InspectFile(ctx, &pfs.InspectFileRequest{stateFile})
+	if err == nil {
+		datumInfo.State = pps.DatumState_SKIPPED
+		// Datum wasn't added in this commit, so it was skipped
 		return datumInfo, nil
+	} else if !strings.Contains(err.Error(), "not found") {
```

Comment:
There should be a helper somewhere that you can use for this. If not it's probably worth it to add one to this file for the one below as well?